File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    const gcd = (a, b) => a % b === 0 ? b : gcd(b, a % b);
    const lcm = (a, b) => a * b / gcd(a, b);
    
    return arr.reduce((acc, val) => {
        return lcm(acc, val);
    }, 1);
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    def gcd(a, b):
        return b if a % b == 0 else gcd(b, a % b)

    def lcm(a, b):
        return a * b // gcd(a, b)

    result = 1
    for val in arr:
        result = lcm(result, val)

    return result
